src/snowflake/snowpark/modin/pandas/indexing.py

```diff
@@ -1049,14 +1049,34 @@ def __getitem__(
 
         original_row_loc = row_loc  # keep a copy for error message
 
+        # IR changes! Add iloc get nodes to AST.
+        stmt = self.df._get_session()._ast_batch.assign()
+        ast = stmt.expr
+        ast.pd_dataframe_iloc_get.df.pd_dataframe_ref.id.bitfield1 = (
+            self.df._get_ast_id()
+        )
+
+        # The row and column locations can be a scalar, slice, list-like, or Series object - record all of these
+        # separately for AST.
```

Comment:
I don't think you should do this here. I would have a reusable function `ast_expr_from_python_val(ast_expr, val)` that attempts to map an arbitrary Python value to an AST expression, following the types in the AST. You would invoke it for every value of type `Expr` in the AST. It will map all the built-ins (functions, scalars, lists, maps, slices, etc.) plus emit `Ref` nodes for dataframes, series, and indexes.

Once you have this fn, you can just blindly pass `key` to it.

Reply:
I think I see what you mean. Is this supposed to happen on the client side/`indexing.py` or in the TcmSession logic?